synapse/lib/autodoc.py

```diff
@@ -186,6 +199,31 @@ def getArgLines(rtype):
 
     return lines
 
+def genDeprecationNote(name, depr, runt=False):
+    assert name is not None
+    assert depr is not None
+    lines = []
+    if runt:
+        lines.append('.. warning::')
+    else:
+        lines.append('Warning:')
+
+    mesg = depr.get('mesg')
+    date = depr.get('eoldate')
+    vers = depr.get('eolvers')
+
+    assert mesg or date or vers
+
+    if mesg:
```

Comment:
If mesg is specified then do we not print the eoldate/vers?  IMO either eoldate/vers should be required, printed, and then the optional message comes after.

Reply:
This is from @vEpiphyte . I took it as an ranked list of "prefer this one, but fall back if you don't have anything else". I don't have a strong leaning one way or another so I'll defer to @vEpiphyte.